### PR TITLE
165675611 admin/user endpoint refactoring

### PR DESCRIPTION
**Pull request by philosotractor**
This initiates the refactoring of admin routes and endpoints to use properly HTTP status codes,  methods/verbs and url parametrization.

# Added
*  components: http.transit-response multi-arity to specifying http status
*  test: http-client/delete helper for unit tests
   
# Changed
* services: admin/user DELETE id parsing for security and test for bad id
 services: admin/user getting users refactored from POST to GET
 services: admin/users refactored to admin/user and to use url parameters
 services: admin/delete-user endpoint changed to DELETE admin/user/:id
 services: tests improved admin/users, inspect https status etc.
 services: admin endpoints refactored admin/users and admin/delete-user

Remaining admin endopoints will be refactored in upcoming PRs

### CODE REVIEW

File: ote/src/clj/ote/services/admin.clj [hunk 1]
```diff
@@ -487,42 +518,39 @@
...
 (define-service-component CSVAdminReports
-  {}
-  ^{:format :csv
-    :filename (str "raportti-" (time/format-date-iso-8601 (time/now)) ".csv")}
-  (GET "/admin/reports/transport-operator/:type"
-       {{:keys [type]} :params
-        user :user}
-    (require-admin-user "reports/transport-operator" (:user user))
-    (transport-operator-report db type)))
+                          {}
```
Comment (rckopteri) [inline ote/src/clj/ote/services/admin.clj]: indention

File: ote/src/cljs/ote/app/controller/admin.cljs [hunk 2]
```diff
@@ -172,10 +174,12 @@
     (assoc app :flash-message-error "Käyttäjän poistaminen epäonnistui"))
 
   SearchUsersResponse
-  (process-event [{response :response} app]
+  (process-event [{response :response :as all} app]
```
Comment (rckopteri) [inline ote/src/cljs/ote/app/controller/admin.cljs]: Is "all" used?

File: ote/src/clj/ote/services/admin.clj (hunk 1 above)
Comment (philosotractor) [reply ote/src/clj/ote/services/admin.clj]: Fix'd in next PR

File: ote/src/cljs/ote/app/controller/admin.cljs (hunk 2 above)
Comment (philosotractor) [reply ote/src/cljs/ote/app/controller/admin.cljs]: Fix'd in next PR

**Comment by philosotractor** [comment]
Backslash in branch name caused a CI deploy issue, this PR closed now and new opened: https://github.com/finnishtransportagency/mmtis-national-access-point/pull/1026

**Pull request closed without merging**